cfme/tests/storage/test_volume.py

```diff
@@ -224,7 +227,30 @@ def test_storage_volume_attached_delete():
             4. check for flash message " Cloud Volume "Volume_name" cannot be
             removed because it is attached to one or more Instances "
         """
-    pass
+    volume = create_volume(appliance, provider, from_manager, az=instance_fixture.
+                           vm_default_args["environment"]["availability_zone"], should_assert=True)
+
+    # attach
+    volume.attach_instance(name=instance_fixture.name, mountpoint='/dev/sdm',
+                           from_manager=from_manager)
+    wait_for(lambda: volume.instance_count == 1, delay=15, timeout=600)
+
+    try:
+        volume.delete(from_manager=from_manager)
+        pytest.fail("Attached volume was deleted!")
+    except Exception:
+        if from_manager:
+            view = volume.browser.create_view(StorageManagerVolumeAllView,
+                                              additional_context={'object': volume.parent.manager})
+        else:
+            view = volume.create_view(VolumeDetailsView)
+        assert view.flash.assert_message('Cloud Volume "{}" cannot be removed because it is '
+                                         'attached to one or more Instances'.format(volume.name))
+    # detach
+    volume.detach_instance(name=instance_fixture.name, from_manager=from_manager)
```

Comment:
I would expect this and the following lines to be part of cleanup, especially if volume creation is moved to a fixture

Reply:
Moved to a fixture.